components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -146,10 +171,18 @@ export class TeamDBImpl implements TeamDB {
             );
         }
 
+        let slug = slugify(name, { lower: true });
         const teamRepo = await this.getTeamRepo();
+
+        const existingTeam = await teamRepo.findOne({ slug, deleted: false, markedDeleted: false });
+        if (!!existingTeam) {
+            slug = slug + "-" + randomBytes(10).toString("hex");
```

Comment:
**thought**: It's feels right we are going down this route, see relevant discussions[[1](https://gitpod.slack.com/archives/C04E8MU39JQ/p1676047161150339?thread_ts=1676036280.889149&cid=C04E8MU39JQ)][[2](https://gitpod.slack.com/archives/C021Q4BL8E8/p1625064176314500?thread_ts=1625063886.313200&cid=C021Q4BL8E8)]. 💭